captum/optim/_param/image/images.py

```diff
@@ -364,6 +354,95 @@ def forward(self) -> torch.Tensor:
         return torch.stack(A).refine_names("B", "C", "H", "W")
 
 
+class SharedImage(ImageParameterization):
+    """
+    Share some image parameters across the batch.
+    Mordvintsev, et al., "Differentiable Image Parameterizations", Distill, 2018.
+    https://distill.pub/2018/differentiable-parameterizations/
+    """
+
+    def __init__(
+        self,
+        shapes: Union[Tuple[Tuple[int]], Tuple[int]] = None,
+        parameterization=None,
+        offset: Union[int, Tuple[int], Tuple[Tuple[int]], None] = None,
+    ) -> None:
+        super().__init__()
+        A = []
+        shared_shapes = [shapes] if type(shapes[0]) is not tuple else shapes
+        for shape in shared_shapes:
+            assert len(shape) >= 2 and len(shape) <= 4
+            shape = ([1] * (4 - len(shape))) + list(shape)
+            batch, channels, height, width = shape[0], shape[1], shape[2], shape[3]
+            A.append(torch.nn.Parameter(torch.randn([batch, channels, height, width])))
+        self.shared_init = torch.nn.ParameterList(A)
+        self.parameterization = parameterization
+        self.offset = self.get_offset(offset, len(A)) if offset is not None else None
+
+    def get_offset(self, offset: Union[int, Tuple[int]], n: int) -> List[List[int]]:
+        if type(offset) is tuple or type(offset) is list:
+            if type(offset[0]) is tuple or type(offset[0]) is list:
+                assert len(offset) == n and all(len(t) == 4 for t in offset)
+            else:
+                assert len(offset) >= 1 and len(offset) <= 4
+                offset = [([0] * (4 - len(offset))) + list(offset)] * n
+        else:
+            offset = [[offset] * 4] * n
+        offset = [[int(o) for o in v] for v in offset]
+        return offset
+
+    def apply_offset(
+        self, x_list: List[torch.Tensor], size: Tuple[int]
+    ) -> List[torch.Tensor]:
+        assert len(size) == 4
+        A = []
+        for x, offset in zip(x_list, self.offset):
+            x = F.pad(x, offset, "reflect")
+            x = x[: size[0], : size[1], : size[2], : size[3]]
+            A.append(x)
+        return A
+
+    def interpolate_tensor(
+        self, x: torch.Tensor, size: InitSize, batch: int, channels: int
```

Comment:
nit: instead of passing size it might make more clear to pass hight and width.

Reply:
Yeah, that would probably make it easier to understand. I'll also order the input variables as BCHW as well.